### ISSUE TITLE
Warning C4819  when building samples

### ISSUE THREAD

**Issue by quicklyslow** [OurMachinery/themachinery-public#619]
There is a warning C4819 treat as error when building samples, should add buildoptions {"/utf-8"} to "samples/premake5.lua" after linkoptions {"/ignore:4099"} to fix it

**Comment by simon-ourmachinery** [comment]
Our CI did not catch this either on Windows nor on Linux. Which OS do you have? Do you still have the issue?

**Comment by quicklyslow** [comment]
Yes, OS is win10 but the language is chs/chinese.

**Comment by simon-ourmachinery** [comment]
Oh dang I did not think about this! I will add it to the template and all the premake files which don't have this

**Comment by simon-ourmachinery** [comment]
fixed see references

**Comment by infosia** [comment]
I hit similar issue when I build the machinery from source. I used Windows 10 configured in Japanese. Seems like `themachinery\foundation\os.linux.c(953,1)` has the issue -

```
themachinery\foundation\os.linux.c(953,1): error C2220: the following warning is treated as an error [themachinery\foundation\build\foundation\vs2019\foundation.vcxproj]
themachinery\foundation\os.linux.c(953,1): warning C4819: The file contains a character that cannot be represented in the current code page (932). Save the file in Unicode format to prevent data loss [themachinery\foundation\build\foundation\vs2019\foundation.vcxproj]
```
 
For me following patch fixes the issue.

```diff
$ git diff foundation/premake5.lua
diff --git a/foundation/premake5.lua b/foundation/premake5.lua
index 1cca0a8c7..3c4b99b48 100644
--- a/foundation/premake5.lua
+++ b/foundation/premake5.lua
@@ -59,6 +59,7 @@ filter { "platforms:Win64" }
     defines { "TM_OS_WINDOWS", "_CRT_SECURE_NO_WARNINGS" }
     staticruntime "On"
     architecture "x64"
+    buildoptions { "/utf-8" }
     disablewarnings {
         "4057", -- Slightly different base types. Converting from type with volatile to without.
         "4100", -- Unused formal parameter. I think unusued parameters are good for documentation.
```

**Comment by niklas-ourmachinery** [comment]
Fixed for 2021.8
